32blit-stm32/Inc/32blit_battery.h

```diff
@@ -7,21 +7,28 @@
 
 #include "engine/running_average.hpp"
 
-struct BatteryInformation {
-    const char * status_text;
-    const char * vbus_text;
-    float voltage;
-
-    // low level status info
-    uint8_t battery_status;
-    uint8_t battery_fault;
+enum BatteryVbusStatus {
+  VbusUnknown,
+  USBHost,
+  AdapterPort,
+  OnTheGo,
 };
 
-enum BatteryStatus {
+enum BatteryChargeStatus {
   NotCharging,
   PreCharging,
   FastCharging,
-  ChargingComplete
+  ChargingComplete,
+};
+
+struct BatteryInformation {
+    BatteryChargeStatus charge_status;
+    const char * charge_text;
+
+    BatteryVbusStatus vbus_status;
+    const char * vbus_text;
+
+    float voltage;
 };
 
 class BatteryState final {
```

Comment:
`namespace battery` instead of a class + a single instance? (Also matches display/sound/gpio.hpp, which I think is the direction the code was heading in)

Reply:
There is currently no namespace used in the firmware part. The engine is namespaced into `blit`. Should we somehow try to make hierarchies here, or just a top level `battery` namespace? 

Or perhaps `blit::firmware::battery` ?

Reply:
There is in the three files I mentioned :smile:

https://github.com/pimoroni/32blit-beta/blob/master/32blit-stm32/Inc/display.hpp#L15
https://github.com/pimoroni/32blit-beta/blob/master/32blit-stm32/Inc/sound.hpp#L16
https://github.com/pimoroni/32blit-beta/blob/master/32blit-stm32/Inc/gpio.hpp

and "soon" https://github.com/Daft-Freak/32blit-beta/blob/minimulti/32blit-stm32/Inc/multiplayer.hpp

Reply:
Yeah I think a `battery` namespace is a nice way to clean things up and it seems - as you suggest - to have emerged as a defacto standard. I think "go with the flow" applies here, even though we've not really sat around and thought about it.